### ISSUE TITLE
Hash mode router-link generated anchor href not a hash url

### ISSUE THREAD

**Issue by tilleps** [vuejs/vue-router#780]
For "hash" being the default mode, shouldn't the generated anchors generate hashed URLs?  (For cases where one would want to copy URL or open link in a new tab/window)

The following `router-link`:

``` html
<router-link to="/example">Example</router-link>
```

Generates:

``` html
<a href="/example">Example</a>
```

Expected:

``` html
<a href="#/example">Example</a>
```

Clicking on the generated (non-hashed) anchor seems to work, it changes the URL to the hashed version of the URL.

**Comment by LinusBorg** [comment]
Should not be too hard, and seems nessessary now that we just added support for "new tab with middle click" again (#758).

**Comment by goncha** [comment]
@tilleps use `base` option as workaround

**Comment by myst729** [comment]
Run into this issue too.

The root cause should be here: https://github.com/vuejs/vue-router/blob/dev/src/components/link.js#L33

**Comment by marvinroger** [comment]
Why was this closed? The latest version does not contain a fix yet.

**Comment by rpkilby** [comment]
@marvinroger - this has been fixed on dev (see #809). A release just hasn't been published with the changes yet.

**Comment by marvinroger** [comment]
Alright, thanks!
